teos/src/bitcoin_cli.rs

```diff
@@ -87,11 +90,21 @@ impl<'a> BitcoindClient<'a> {
             rpc_password,
         };
 
-        // Test that bitcoind is reachable
-        match client.get_best_block_hash_and_height().await {
-            Ok(_) => Ok(client),
-            Err(e) => Err(e),
+        // Test that bitcoind is reachable.
+        let btc_network = client.get_chain().await?;
+
+        // Assert teos runs on the same chain/network as bitcoind.
+        if btc_network != teos_network {
+            return Err(Error::new(
+                ErrorKind::InvalidInput,
+                format!(
+                    "bitcoind is running on {} but teosd is set to run on {}",
+                    btc_network, teos_network
+                ),
+            ));
         }
+
+        Ok(client)
```

Comment:
```suggestion
        if client.get_chain().await? != teos_network {
            Err(Error::new(
                ErrorKind::InvalidInput,
                format!(
                    "bitcoind is running on {} but teosd is set to run on {}",
                    btc_network, teos_network
                ),
            ))
        } else {
            Ok(client)
        }
```

Reply:
It's the same logic :question:

Reply:
It should be, but as you mentioned the temp variable is needed for the logging

Reply:
idk, I feel it's better to have all the checks above then `Ok(something)` at the end rather than integrating `Ok(something)` in the last check. But no major difference really. let me change this quickly.